modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2783,6 +2787,9 @@ def compute_groupby(df, drop=False, partition_idx=0):
                     )
                     result.index.names = new_index_names
 
+                if len(misaggregated_cols) > 0:
```

Comment:
This won't work for numpy-like arrays (e.g. pandas Index) since its `__nonzero__` raises an exception:

<details><summary>Example</summary>

```python
>>> if np.array([1, 2]):
...     print("some")
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
>>> if np.array([]):
...     print("some")
...
__main__:1: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
>>> if pandas.Index([1, 2]):
...     print("some")
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "lib\site-packages\pandas\core\indexes\base.py", line 2810, in __nonzero__
    f"The truth value of a {type(self).__name__} is ambiguous. "
ValueError: The truth value of a Int64Index is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
>>> if pandas.Index([]):
...     print("some")
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "lib\site-packages\pandas\core\indexes\base.py", line 2810, in __nonzero__
    f"The truth value of a {type(self).__name__} is ambiguous. "
ValueError: The truth value of a Index is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

</details>